File: Client/app/src/main/java/com/example/apekshakhandelwal/client/customgrid.java
package com.example.apekshakhandelwal.client;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

/**
 * Created by Apeksha Khandelwal on 19-06-2017.
 */

public class customgrid extends BaseAdapter {


    Context context;
    int[] imagesid;
    String[] namesid;

    public customgrid(Context c,String[] names, int[] images) {
        context = c;
        for(int i=0;i<names.length;i++) {
            imagesid[i] = images[i];
            namesid[i] = names[i];
        }
    }

    @Override
    public int getCount() {
        return namesid.length;
    }

    @Override
    public Object getItem(int position) {
        return null;
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent)
    {
        // TODO Auto-generated method stub
        View grid;
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);

        if (convertView == null)
        {
            grid = new View(context);
            grid = inflater.inflate(R.layout.activity_grid, null);
            TextView textView = (TextView) grid.findViewById(R.id.text);
            ImageView imageView = (ImageView)grid.findViewById(R.id.image);
            textView.setText(namesid[position]);
            imageView.setImageResource(imagesid[position]);
        }

        else
            grid = convertView;

        return grid;
    }

}

File: Admin_Jovial/am3y/app/src/main/java/com/example/am3y/CustomGrid.kt
package com.example.am3y

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import com.example.am3y.R

class CustomGrid(internal var context: Context?, names: Array<String>, images: IntArray) :
        BaseAdapter() {
    internal var imagesid: IntArray = kotlin.IntArray(images.size)
    internal var namesid: Array<String?> = kotlin.arrayOfNulls<kotlin.String>(names.size)

    init {
        for (i in names.indices) {
            imagesid.set(i, images[i])
            namesid.set(i, names[i])
        }
    }

    override fun getCount(): Int {
        return namesid.size
    }

    override fun getItem(position: Int): Any? {
        return namesid[position]
    }

    override fun getItemId(position: Int): Long {
        return namesid[position] as Long
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        // TODO Auto-generated method stub
        var grid: View
        val inflater = context?.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

        if (convertView == null) {
            grid = inflater.inflate(R.layout.activity_grid, parent,false)
            val textView = grid.findViewById(R.id.text) as TextView
            val imageView = grid.findViewById(R.id.image) as ImageView
            textView.text = namesid[position]
            imageView.setImageResource(imagesid[position])
            System.out.println("CustomGrid")
        } else
            grid = convertView

        return grid
    }
}
